packages/amazonq/src/lsp/client.ts

```diff
@@ -72,6 +74,47 @@ export async function startLanguageServer(
     const clientOptions: LanguageClientOptions = {
         // Register the server for json documents
         documentSelector,
+        middleware: {
+            workspace: {
+                /**
+                 * Convert VSCode settings format to be compatible with flare's configs
+                 */
+                configuration: async (params, token, next) => {
+                    const config = await next(params, token)
+                    if (params.items[0].section === 'aws.q') {
+                        const customization = undefinedIfEmpty(getSelectedCustomization().arn)
+                        /**
+                         * IMPORTANT: This object is parsed by the following code in the language server, **so
+                         * it must match that expected shape**.
+                         * https://github.com/aws/language-servers/blob/1d2ca018f2248106690438b860d40a7ee67ac728/server/aws-lsp-codewhisperer/src/shared/amazonQServiceManager/configurationUtils.ts#L114
+                         */
+                        return [
+                            {
+                                customization,
+                                optOutTelemetry: getOptOutPreference() === 'OPTOUT',
```

Comment:
oh wait this is a boolean? the type they have in the interface vs what is expected to be passed in is so confusing 😅

Reply:
Yeah its weird lol. They expect a boolean then convert that back in to an 'OPTIN' | 'OPTOUT' string:
https://github.com/aws/language-servers/blob/c367ef3a1374032dace0e6755eaa43a1fae6e3c4/server/aws-lsp-codewhisperer/src/shared/amazonQServiceManager/configurationUtils.ts#L117